spark/v3.1/build.gradle

```diff
@@ -213,6 +213,9 @@ project(':iceberg-spark:iceberg-spark-runtime-3.1_2.12') {
     implementation(project(':iceberg-nessie')) {
       exclude group: 'com.google.code.findbugs', module: 'jsr305'
     }
+    implementation (project(':iceberg-snowflake')) {
+      exclude group: 'net.snowflake' , module: 'snowflake-jdbc'
```

Comment:
Before we can include this in runtime Jars, we need to validate the file diff between the resulting Jars, update LICENSE/NOTICE files, and also take a look at the dependency tree to make sure it isn't too broad.

Reply:
Now that we've removed dbutils entirely, I've confirmed that only the newly added classes are the sources added in this PR.

    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ ls -l ./spark/v3.*/spark-runtime/build/libs/*dirty.jar
    -rw-rw-r-- 1 dhuo dhuo 26500156 Dec 20 23:53 ./spark/v3.1/spark-runtime/build/libs/iceberg-spark-runtime-3.1_2.12-b3a2842.dirty.jar
    -rw-rw-r-- 1 dhuo dhuo 26968503 Dec 20 23:53 ./spark/v3.2/spark-runtime/build/libs/iceberg-spark-runtime-3.2_2.12-b3a2842.dirty.jar
    -rw-rw-r-- 1 dhuo dhuo 27008357 Dec 20 23:53 ./spark/v3.3/spark-runtime/build/libs/iceberg-spark-runtime-3.3_2.12-b3a2842.dirty.jar
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ jar tf ./spark/v3.1/spark-runtime/build/libs/iceberg-spark-runtime-3.1_2.12-b3a2842.dirty.jar > sparkclasses-3.1.before.txt
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ jar tf ./spark/v3.2/spark-runtime/build/libs/iceberg-spark-runtime-3.2_2.12-b3a2842.dirty.jar > sparkclasses-3.2.before.txt
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ jar tf ./spark/v3.3/spark-runtime/build/libs/iceberg-spark-runtime-3.3_2.12-b3a2842.dirty.jar > sparkclasses-3.3.before.txt
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ ls -l ./spark/v3.*/spark-runtime/build/libs/*dirty.jar
    -rw-rw-r-- 1 dhuo dhuo 26524572 Dec 21 00:05 ./spark/v3.1/spark-runtime/build/libs/iceberg-spark-runtime-3.1_2.12-676d024.dirty.jar
    -rw-rw-r-- 1 dhuo dhuo 26992919 Dec 21 00:05 ./spark/v3.2/spark-runtime/build/libs/iceberg-spark-runtime-3.2_2.12-676d024.dirty.jar
    -rw-rw-r-- 1 dhuo dhuo 27032773 Dec 21 00:05 ./spark/v3.3/spark-runtime/build/libs/iceberg-spark-runtime-3.3_2.12-676d024.dirty.jar
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ jar tf ./spark/v3.1/spark-runtime/build/libs/iceberg-spark-runtime-3.1_2.12-676d024.dirty.jar > sparkclasses-3.1.after.txt
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ jar tf ./spark/v3.2/spark-runtime/build/libs/iceberg-spark-runtime-3.2_2.12-676d024.dirty.jar > sparkclasses-3.2.after.txt
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ jar tf ./spark/v3.3/spark-runtime/build/libs/iceberg-spark-runtime-3.3_2.12-676d024.dirty.jar > sparkclasses-3.3.after.txt
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ diff sparkclasses-3.1.before.txt sparkclasses-3.1.after.txt
    421a422,437
    > org/apache/iceberg/snowflake/
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient$1.class
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient$QueryHarness.class
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient$ResultSetParser.class
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient.class
    > org/apache/iceberg/snowflake/NamespaceHelpers$1.class
    > org/apache/iceberg/snowflake/NamespaceHelpers.class
    > org/apache/iceberg/snowflake/SnowflakeCatalog$1.class
    > org/apache/iceberg/snowflake/SnowflakeCatalog$FileIOFactory.class
    > org/apache/iceberg/snowflake/SnowflakeCatalog.class
    > org/apache/iceberg/snowflake/SnowflakeClient.class
    > org/apache/iceberg/snowflake/SnowflakeIdentifier$1.class
    > org/apache/iceberg/snowflake/SnowflakeIdentifier$Type.class
    > org/apache/iceberg/snowflake/SnowflakeIdentifier.class
    > org/apache/iceberg/snowflake/SnowflakeTableMetadata.class
    > org/apache/iceberg/snowflake/SnowflakeTableOperations.class
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ diff sparkclasses-3.2.before.txt sparkclasses-3.2.after.txt
    486a487,502
    > org/apache/iceberg/snowflake/
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient$1.class
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient$QueryHarness.class
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient$ResultSetParser.class
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient.class
    > org/apache/iceberg/snowflake/NamespaceHelpers$1.class
    > org/apache/iceberg/snowflake/NamespaceHelpers.class
    > org/apache/iceberg/snowflake/SnowflakeCatalog$1.class
    > org/apache/iceberg/snowflake/SnowflakeCatalog$FileIOFactory.class
    > org/apache/iceberg/snowflake/SnowflakeCatalog.class
    > org/apache/iceberg/snowflake/SnowflakeClient.class
    > org/apache/iceberg/snowflake/SnowflakeIdentifier$1.class
    > org/apache/iceberg/snowflake/SnowflakeIdentifier$Type.class
    > org/apache/iceberg/snowflake/SnowflakeIdentifier.class
    > org/apache/iceberg/snowflake/SnowflakeTableMetadata.class
    > org/apache/iceberg/snowflake/SnowflakeTableOperations.class
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ diff sparkclasses-3.3.before.txt sparkclasses-3.3.after.txt
    498a499,514
    > org/apache/iceberg/snowflake/
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient$1.class
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient$QueryHarness.class
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient$ResultSetParser.class
    > org/apache/iceberg/snowflake/JdbcSnowflakeClient.class
    > org/apache/iceberg/snowflake/NamespaceHelpers$1.class
    > org/apache/iceberg/snowflake/NamespaceHelpers.class
    > org/apache/iceberg/snowflake/SnowflakeCatalog$1.class
    > org/apache/iceberg/snowflake/SnowflakeCatalog$FileIOFactory.class
    > org/apache/iceberg/snowflake/SnowflakeCatalog.class
    > org/apache/iceberg/snowflake/SnowflakeClient.class
    > org/apache/iceberg/snowflake/SnowflakeIdentifier$1.class
    > org/apache/iceberg/snowflake/SnowflakeIdentifier$Type.class
    > org/apache/iceberg/snowflake/SnowflakeIdentifier.class
    > org/apache/iceberg/snowflake/SnowflakeTableMetadata.class
    > org/apache/iceberg/snowflake/SnowflakeTableOperations.class